separatedunit.cpp

```diff
@@ -158,6 +153,24 @@ void SeparatedUnit::Cohere(SeparatedUnit& toCopy)
     }
 }
 
+void SeparatedUnit::Cohere(std::vector<std::shared_ptr<CoherentUnit>> toCopy)
+{
+    bitLenInt i;
+    bitLenInt toAddCuLen = toCopy.size();
+    for (i = 0; i < toAddCuLen; i++) {
+        Cohere(*(toCopy[i]));
+    }
+}
+
+void SeparatedUnit::Cohere(std::vector<std::shared_ptr<SeparatedUnit>> toCopy)
```

Comment:
There’s not any code testing any of these Cohere(SU) (or copy constructor) code blocks.

Reply:
Note: non-blocking on this PR, but should be addressed soon.